src/Compilers/CSharp/Test/Emit/CodeGen/UnsafeTests.cs

```diff
@@ -5536,16 +5537,37 @@ void Test(int i, uint u, long l, ulong ul)
             p = p + u;
             p = p + l;
             p = p + ul;
+            p = p + (-2);
```

Comment:
>p = p + (-2); [](start = 12, length = 13)

It would be good to have tests for runtime behavior for scenarios that we are fixing.